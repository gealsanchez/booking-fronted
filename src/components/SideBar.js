import React from 'react';
import { Link } from 'react-router-dom';
import logo from '../images/logo.png';

const SideBar = () => (
  <nav className="sidebar">
    <div className="sidebar-header">
      <div className="sidebar-header-icon">
        <img className="logo" src={logo} alt="Our logo" />
      </div>
    </div>
    <ul className="sidebar-list">
      <li className="sidebar-item">
        <Link to="/main" className="link">AUTOMOBILES</Link>
      </li>
      <li className="sidebar-item">
        <Link to="/add_automobile" className="link">ADD AUTOMOBILE</Link>
      </li>
      <li className="sidebar-item">
        <Link to="/automobiles/delete" className="link">DELETE AUTOMOBILE</Link>
      </li>
      <li className="sidebar-item">
        <Link to="/reserve" className="link">RESERVE</Link>
      </li>
    </ul>
    <Link to="/signin" className="link sign-out-side">Sign Out</Link>
  </nav>
);

export default SideBar;
