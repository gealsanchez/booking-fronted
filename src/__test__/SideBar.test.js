import React from 'react';
import { Provider } from 'react-redux';
import { render } from '@testing-library/react';
import { BrowserRouter as Router } from 'react-router-dom';
import store from '../redux/configureStore';
import SideBar from '../components/SideBar';

it('SideBar page should render correctly', () => {
  const result = render(
    <React.StrictMode>
      <Router>
        <Provider store={store}>
          <SideBar />
        </Provider>
      </Router>
    </React.StrictMode>,
  );
  expect(result).toMatchSnapshot();
});
