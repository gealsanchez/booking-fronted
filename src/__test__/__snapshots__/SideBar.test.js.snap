// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SideBar page should render correctly 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <nav
        class="sidebar"
      >
        <div
          class="sidebar-header"
        />
        <ul
          class="sidebar-list"
        >
          <li
            class="sidebar-item"
          >
            <a
              class="link"
              href="/main"
            >
              AUTOMOBILES
            </a>
          </li>
          <li
            class="sidebar-item"
          >
            <a
              class="link"
              href="/add_automobile"
            >
              ADD AUTOMOBILE
            </a>
          </li>
          <li
            class="sidebar-item"
          >
            <a
              class="link"
              href="/automobiles/delete"
            >
              DELETE AUTOMOBILE
            </a>
          </li>
          <li
            class="sidebar-item"
          >
            <a
              class="link"
              href="/tickets"
            >
              Ticket
            </a>
          </li>
        </ul>
        <button
          class="sign-out-side"
          type="button"
        >
          <span>
            Sign Out
          </span>
        </button>
      </nav>
    </div>
  </body>,
  "container": <div>
    <nav
      class="sidebar"
    >
      <div
        class="sidebar-header"
      />
      <ul
        class="sidebar-list"
      >
        <li
          class="sidebar-item"
        >
          <a
            class="link"
            href="/main"
          >
            AUTOMOBILES
          </a>
        </li>
        <li
          class="sidebar-item"
        >
          <a
            class="link"
            href="/add_automobile"
          >
            ADD AUTOMOBILE
          </a>
        </li>
        <li
          class="sidebar-item"
        >
          <a
            class="link"
            href="/automobiles/delete"
          >
            DELETE AUTOMOBILE
          </a>
        </li>
        <li
          class="sidebar-item"
        >
          <a
            class="link"
            href="/tickets"
          >
            Ticket
          </a>
        </li>
      </ul>
      <button
        class="sign-out-side"
        type="button"
      >
        <span>
          Sign Out
        </span>
      </button>
    </nav>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
